lib/migrator/helpers.ts

```diff
@@ -198,3 +199,77 @@ export const getTargetBlockDevice = async (mountLabel: string = 'C') => {
 		keepOriginal: true
 	})
 }
+
+/** 
+ * Identifies the type of WiFi authentication for a connection profile.
+ * NONE means no authentication.
+ */
+export const enum WifiAuthType { NONE, WPA2_PSK, WPA3_SAE }
+
+/** Configuration values for a single WiFi network profile. */
+export interface ConnectionProfile {
+	name: string;
+	wifiSsid: string;
+	wifiAuthType: WifiAuthType;
+	wifiKey: string;
+}
+
+/** Generates the contents for a balenaOS network configuration file. */
+const generateWifiConfig = (profile: ConnectionProfile): string => {
+	const connectionSection = `[connection]
+id=${profile.name}
+type=wifi
+`
+	const wifiSection = `
+[wifi]
+mode=infrastructure
+ssid=${profile.wifiSsid}
+`
+	let keyMgmt
+	switch (profile.wifiAuthType) {
+		case WifiAuthType.NONE:
+			keyMgmt = 'none'
+			break
+		case WifiAuthType.WPA2_PSK:
+			keyMgmt = 'wpa-psk'
+			break
+		case WifiAuthType.WPA3_SAE:
+			keyMgmt = 'sae'
+			throw Error("WifiAuthType WPA3_SAE not supported")
+		default:
+			throw Error(`WifiAuthType ${profile.wifiAuthType} not defined`)
+	}
+		const wifiSecuritySection = `
+[wifi-security]
+auth-alg=open
+key-mgmt=${keyMgmt}
+psk=${profile.wifiKey}
+`
+
+	const ipSection = `
+[ipv4]
+method=auto
+
+[ipv6]
+addr-gen-mode=stable-privacy
+method=auto
+`
+
+	return connectionSection.concat(wifiSection, wifiSecuritySection, ipSection)
+}
+
+/**
+ * Writes a balenaOS network configuration file for the provided network connection 
```

Comment:
Same -- What's about this function that makes it balenaOS-specific?

Reply:
Got confirmation that the configuration file is pure NetworkManager.